ListLibrary/MyList.cs

```diff
@@ -0,0 +1,519 @@
+            {
+                if (i == pos)
+                {
+                    _items[i] = item;
+                    break;
+                }
+                else
+                {
+                    _items[i] = _items[i - 1];
+                }
+            }
+
+            _size++;
+        }
+
+        public void Add(MyList<T> items)
+        {
+            AddByIndex(_size, items);
+        }
+
+        public void AddFront(MyList<T> items)
+        {
+            AddByIndex(0, items);
+        }
+
+        public void AddByIndex(int pos, MyList<T> items)
+        {
+            var newSize = _size + items.Count;
+
+            if (newSize >= _items.Length)
+            {
+                UpdateCapacity(newSize);
+            }
+
+            _size += items.Count;
+
+            for (int i = _size - 1, j = items.Count - 1; j >= 0; i--)
+            {
+                if (i == pos + j)
+                {
+                    _items[i] = items[j];
+                    j--;
+                }
+                else
+                {
+                    _items[i] = _items[i - items.Count];
+                }
+            }
+        }
+
+        #endregion
+
+        #region Remove
+
+        public void RemoveOneFromEnd()
+        {
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            _size--;
+        }
+
+        public void RemoveOneFromStart()
+        {
+            if (_size == 0)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            _size--;
+
+            for (int i = 0; i < _size; i++)
+            {
+                _items[i] = _items[i + 1];
+            }
+        }
+
+        public void RemoveAt(int index)
+        {
+            if (_size == 0 || index < 0 || index > _size)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            _size--;
+
+            for (int i = 0; i < _size; i++)
+            {
+                if (i >= index)
+                {
+                    _items[i] = _items[i + 1];
+                }
+            }
+        }
+
+        public void RemoveNFromEnd(int quantity)
+        {
+            if (_size < quantity)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            _size -= quantity;
+        }
+
+        public void RemoveNFromStart(int quantity)
+        {
+            if (_size < quantity)
+            {
+                throw new ArgumentOutOfRangeException("Size");
+            }
+
+            _size -= quantity;
+
+            for (int i = 0; i < _size; i++)
+            {
+                _items[i] = _items[i + quantity];
+            }
+        }
+
+        public void RemoveNFromIndex(int index, int quantity)
+        {
+            if (_size < quantity + index || index < 0 || index + quantity > _size)
```

Comment:
Two exceptions.
And check how deletion works in default `List`. Maybe it's ok to delete more than exist (and check same logic for single remove)

Reply:
Default List throw exception if you try to delete more than exist. I'll keep my realisation. Exceptions fixed